test_conformance/api/test_semaphores.cpp

```diff
@@ -0,0 +1,1137 @@
+    test_error(err, "Could not create user event");
+
+    // Signal semaphore (dependency on user_event_1)
+    clEventWrapper signal_1_event;
+    err = clEnqueueSignalSemaphoresKHR(queue, 1, &sema, nullptr, 1,
+                                       &user_event_1, &signal_1_event);
+    test_error(err, "Could not signal semaphore");
+
+    // Signal semaphore (dependency on user_event_2)
+    clEventWrapper signal_2_event;
+    err = clEnqueueSignalSemaphoresKHR(queue, 1, &sema, nullptr, 1,
+                                       &user_event_2, &signal_2_event);
+    test_error(err, "Could not signal semaphore");
+
+    // Wait semaphore (dependency on user_event_3)
+    clEventWrapper wait_1_event;
+    err = clEnqueueWaitSemaphoresKHR(queue, 1, &sema, nullptr, 1, &user_event_3,
+                                     &wait_1_event);
+    test_error(err, "Could not wait semaphore");
+
+    // Wait semaphore (dependency on user_event_2)
+    clEventWrapper wait_2_event;
+    err = clEnqueueWaitSemaphoresKHR(queue, 1, &sema, nullptr, 1, &user_event_2,
+                                     &wait_2_event);
+    test_error(err, "Could not wait semaphore");
+
+    // Complete user_event_2
+    err = clSetUserEventStatus(user_event_2, CL_COMPLETE);
+    test_error(err, "Could not set user event to CL_COMPLETE");
+
+    // Flush and delay
+    err = clFlush(queue);
+    test_error(err, "Could not flush queue");
+    std::this_thread::sleep_for(std::chrono::seconds(FLUSH_DELAY_S));
+
+    // Ensure only second signal and second wait completed
+    test_assert_event_inprogress(signal_1_event);
+    test_assert_event_complete(signal_2_event);
+    test_assert_event_inprogress(wait_1_event);
+    test_assert_event_complete(wait_2_event);
+
+    // Complete user_event_1
+    err = clSetUserEventStatus(user_event_1, CL_COMPLETE);
+    test_error(err, "Could not set user event to CL_COMPLETE");
+
+    // Complete user_event_3
+    err = clSetUserEventStatus(user_event_3, CL_COMPLETE);
+    test_error(err, "Could not set user event to CL_COMPLETE");
+
+    // Finish
+    err = clFinish(queue);
+    test_error(err, "Could not finish queue");
+
+    // Ensure all events are completed
+    test_assert_event_complete(signal_1_event);
+    test_assert_event_complete(signal_2_event);
+    test_assert_event_complete(wait_1_event);
+    test_assert_event_complete(wait_2_event);
+
+    // Release semaphore
+    err = clReleaseSemaphoreKHR(sema);
+    test_error(err, "Could not release semaphore");
+
+    return TEST_PASS;
+}
+
+// Test it is possible to export a semaphore to a sync fd and import the same
+// sync fd to a new semaphore
+int test_semaphores_import_export_fd(cl_device_id deviceID, cl_context context,
+                                     cl_command_queue defaultQueue,
+                                     int num_elements)
+{
+    cl_int err;
+
+    if (!is_extension_available(deviceID, "cl_khr_semaphore"))
```

Comment:
recommend testing for "cl_khr_external_semaphore_sync_fd"

Reply:
Ack